### PR TITLE
JSON-ize slot machines

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Infrastructure "JSON-ize slot machines"

#### Purpose of change
Slot machines are pretty neat things and could easily be expanded upon for casino-like areas and NPC bases. The main problem with them is the barrier to entry, as their functions were contained within C++. Additionally, it didn't seem that the full scope of their functions worked properly, so this changes that by converting the code to a JSON format. [Maleclypse also wanted this to happen](https://discord.com/channels/598523535169945603/598529174302490644/1176597552238440578).

#### Describe the solution
Remove the old slot machine examine action and add a new JSON format to it, with EOC's as the primary way in which it works. I also decided to expand the betting amounts to include twenty-, five-, and one-dollar amounts. You can pick which one you want with a talk topic, that then runs an EOC to determine if you won, and if so, how much. I made sure to try and keep these percentages consistent with what the old C++ had for the `one-in` chances. The current possible results are three cherries (gives a refund), three bells (doubles your money), three stars (triples your money), and a jackpot (pays out 100x your money). 

#### Describe alternatives you've considered
Not doing this, or expanding the functionality in C++

#### Testing
Went to a casino and lost all of my money, with no issues.

#### Additional context
I'm not too familiar with how slot machines work, so I didn't worry too much about following with the patterns of actual games. Maleclypse also said that they wanted to do this, so I'll leave that in their hands.

### CODE REVIEW

File: data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        ]
+      ]
+    }
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "slot_machine_win_payout",
+    "condition": { "math": [ "u_bet_amount", "==", "20" ] },
+    "effect": [
+      { "u_spawn_item": "money_twenty", "count": { "global_val": "gamble_multiplier" } },
```
Comment (MNG-cataclysm) [inline data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json:128]:
```suggestion
      { "u_spawn_item": "money_twenty", "count": { "global_val": "gamble_multiplier", "default": 1 } },
```

File: data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      { "math": [ "gamble_multiplier", "=", "0" ] },
+      { "math": [ "u_bet_amount", "=", "0" ] }
+    ],
+    "false_effect": {
+      "run_eocs": {
+        "id": "win_10",
+        "condition": { "math": [ "u_bet_amount", "==", "10" ] },
+        "effect": [
+          { "u_spawn_item": "money_ten", "count": { "global_val": "gamble_multiplier" } },
```
Comment (MNG-cataclysm) [inline data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json:137]:
```suggestion
          { "u_spawn_item": "money_ten", "count": { "global_val": "gamble_multiplier", "default": 1 } },
```

File: data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          { "math": [ "gamble_multiplier", "=", "0" ] },
+          { "math": [ "u_bet_amount", "=", "0" ] }
+        ],
+        "false_effect": {
+          "run_eocs": {
+            "id": "win_5",
+            "condition": { "math": [ "u_bet_amount", "==", "5" ] },
+            "effect": [
+              { "u_spawn_item": "money_five", "count": { "global_val": "gamble_multiplier" } },
```
Comment (MNG-cataclysm) [inline data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json:146]:
```suggestion
              { "u_spawn_item": "money_five", "count": { "global_val": "gamble_multiplier", "default": 1 } },
```

File: data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+              { "math": [ "gamble_multiplier", "=", "0" ] },
+              { "math": [ "u_bet_amount", "=", "0" ] }
+            ],
+            "false_effect": {
+              "run_eocs": {
+                "id": "win_1",
+                "condition": { "math": [ "u_bet_amount", "==", "1" ] },
+                "effect": [
+                  { "u_spawn_item": "money_one", "count": { "global_val": "gamble_multiplier" } },
```
Comment (MNG-cataclysm) [inline data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json:155]:
```suggestion
                  { "u_spawn_item": "money_one", "count": { "global_val": "gamble_multiplier", "default": 1 } },
```

File: src/iexamine.cpp [hunk 5]
```diff
@@ -1716,43 +1716,7 @@ void iexamine::pit_covered( Character &you, const tripoint &examp )
 /**
  * Loop prompt to bet $10.
  */
```
Comment (Maleclypse) [inline src/iexamine.cpp:1718]:
```suggestion
```
Removing this as it applies to what you are removing and not the 2x4 covered pit above.

File: data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                  { "math": [ "u_bet_amount", "=", "0" ] }
+                ]
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+]
```
Comment (Maleclypse) [inline data/json/effects_on_condition/mapgen_eocs/furniture_eocs.json:166]:
```suggestion
]

```

**Pull request merged**